agent/task.go

```diff
@@ -70,6 +70,8 @@ func (a *Agent) startTask(ctx context.Context, tc *taskContext, complete chan<-
 		complete <- evergreen.TaskFailed
 		return
 	}
+	taskConfig.Redacted = tc.expVars.PrivateVars
```

Comment:
is there any way that we're sending private variables in a way that changes our contract about private variables?

Reply:
Nope, the only change here is that we're retrieving them earlier. The change here is to not make the request twice